File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/go_solutions/1041-robot-bounded-in-circle.go
func isRobotBounded(instructions string) bool {
    // Initial position and direction
    x, y := 0, 0
    // 0: North, 1: East, 2: South, 3: West
    dir := 0 

    // dx, dy arrays for movement based on direction
    //      N  E

File: MATTHEW/leetcode/AIcoding/4.GPT-4o/rust_solutions/1041-robot-bounded-in-circle.rs
impl Solution {
    pub fn is_robot_bounded(instructions: String) -> bool {
        let mut x = 0;
        let mut y = 0;
        let mut dir = 0; // 0: North, 1: East, 2: South, 3: West

        for c in instructions.chars() {
            match c {
                'G' => {
                    match dir {
                        0 => y += 1,
                        1 => x += 1,
                        2 => y -= 1,
                        3 => x -= 1,
                        _ => {}
                    }
                }
                'L' => dir = (dir + 3) % 4,
                'R' => dir = (dir + 1) % 4,
                _ => {}
            }
        }

        (x == 0 && y == 0) || dir != 0
    }
}